docs/next/components/mdx/RenderedDAG.tsx

```diff
@@ -0,0 +1,381 @@
+  const [content, setContent] = React.useState('');
+
+  React.useEffect(() => {
+    const load = async () => {
+      const result = await fetch(svgSrc);
+      const text = await result.text();
+      setContent(text);
+    };
+    load();
+  }, [svgSrc]);
+
+  return (
+    <div
+      className="text-black hidden lg:block"
+      style={{
+        minHeight: 400,
+        width: '40.5%',
+        maxWidth: MAX_HOVER_WIDTH,
+        boxSizing: 'border-box',
+        flexShrink: 0,
+      }}
+    >
+      <div
+        onMouseEnter={() => setHover(true)}
+        onMouseLeave={() => setHover(false)}
+        style={{
+          lineHeight: 1.1,
+          position: 'absolute',
+          bottom: 0, // 20
+          top: 0, // 20
+          right: 0,
+          paddingLeft: focus ? 0 : hover ? 5 : 15,
+          width: focus ? '100%' : '40.5%',
+          maxWidth: focus ? '100%' : MAX_HOVER_WIDTH,
+          left: 'unset',
+          cursor: focus ? 'grab' : 'zoom-in',
+          zIndex: focus || hover ? 2 : 1,
+          transition:
+            'width 300ms ease-in-out, max-width 300ms ease-in-out, padding-left 300ms ease-in-out, left 300ms ease-in-out',
+        }}
+      >
+        {/* {isAssetGraph && <MaterializeButton />} */}
+        {focus && (
+          <div
+            onClick={() => setFocus(false)}
+            className="absolute top-4 left-4 flex gap-1 bg-white border border-gray-200 hover:bg-gray-100 transition text-black py-2 px-4 z-50 rounded-full cursor-pointer"
+          >
+            <svg
+              xmlns="http://www.w3.org/2000/svg"
+              height="20px"
+              viewBox="0 0 24 24"
+              width="20px"
+              fill="#000"
+            >
+              <path d="M0 0h24v24H0z" fill="none"></path>
+              <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path>
+            </svg>
+            Close
+          </div>
+        )}
+
+        {content ? (
+          <SVGViewport
+            graphWidth={Number(content.match(/ width=\"([0-9\.]+)\"/)![1])}
+            graphHeight={Number(content.match(/ height=\"([0-9\.]+)\"/)![1])}
+            maxZoom={1.2}
+            maxAutocenterZoom={1.0}
+            onFocus={() => setFocus(true)}
+            onBlur={() => setFocus(false)}
+            content={content}
+          />
+        ) : (
+          <div style={SVGViewportStyles} />
+        )}
+      </div>
+    </div>
+  );
+};
+
+interface SVGViewportProps {
+  content: string;
+  graphWidth: number;
+  graphHeight: number;
+  maxZoom: number;
+  maxAutocenterZoom: number;
+  onFocus: () => void;
+  onBlur: () => void;
+}
+
+interface SVGViewportState {
+  x: number;
+  y: number;
+  scale: number;
+  minScale: number;
+}
+
+interface Point {
+  x: number;
+  y: number;
+}
+
+const MIN_ZOOM = 0.17;
+
+export class SVGViewport extends React.Component<SVGViewportProps, SVGViewportState> {
```

Comment:
Is this a brand new component, or copied from somewhere? If it's new, we shouldn't really be using `React.Component` at this point, and all of this should be achievable with a function component.

Reply:
Ah, yeah, it's copied from https://github.com/dagster-io/dagster-website/blob/master/components/RenderedDAG.tsx, as with most of this component.